### PR TITLE
Plug in rate limit service URL into xds cluster

### PR DESCRIPTION
**Pull request by arkodg**
Signed-off-by: Arko Dasgupta <arko@tetrate.io>

### CODE REVIEW

File: internal/infrastructure/kubernetes/deployment.go [hunk 1]
```diff
@@ -401,7 +401,7 @@ func expectedRateLimitContainers(infra *ir.RateLimitInfra) []corev1.Container {
 				},
 				{
 					Name:  "LOG_LEVEL",
-					Value: "info",
+					Value: "debug",
```
Comment (danehans) [inline internal/infrastructure/kubernetes/deployment.go:404]: Why is this log level now debug?

Comment (arkodg) [reply internal/infrastructure/kubernetes/deployment.go:404]: adds verbosity for this new feature, which can help triaging future issues, can move it to `info` if that is the preference for default

Comment (arkodg) [reply internal/infrastructure/kubernetes/deployment.go]: reverted to `info`

File: internal/infrastructure/kubernetes/service.go [hunk 2]
```diff
@@ -151,6 +152,11 @@ func (i *Infra) expectedRateLimitService(_ *ir.RateLimitInfra) *corev1.Service {
 	return svc
 }
 
+// GetRateLimitServiceURL returns the URL for the rate limit service.
+func GetRateLimitServiceURL(namespace string) string {
+	return "grpc://" + rateLimitInfraName + "." + namespace + ".svc.cluster.local:" + strconv.Itoa(rateLimitInfraGRPCPort)
```
Comment (zirain) [inline internal/infrastructure/kubernetes/service.go:157]: prefer to user `fmt.Printf`

Comment (arkodg) [reply internal/infrastructure/kubernetes/service.go]: ack

File: internal/infrastructure/kubernetes/deployment.go (hunk 1 above)
Comment (danehans) [reply internal/infrastructure/kubernetes/deployment.go]: Thanks for reverting to info. I think it makes sense to have a consistent logging level across EG components.

**Review by danehans — APPROVED** [review]

**Pull request merged**